docking.js

```diff
@@ -618,12 +618,11 @@ var DockedDash = GObject.registerClass({
                 this._ignoreHover = false;
                 // Do not hide if autohide is enabled and mouse is hover
                 if (!this._box.hover || !this._autohideIsEnabled)
-                    this._animateOut(settings.get_double('animation-time'), 0);
+                    this._hide();
             }
             else {
                 this._ignoreHover = true;
-                this._removeAnimations();
-                this._animateIn(settings.get_double('animation-time'), 0);
+                this._show();
```

Comment:
Use `_hide` and `_show` instead to improve readability and keep control over transition behaviours. Only thing is, it now uses the extension's animation delay settings. I don't think it messes with any other setup, but someone with more experience of dash-to-dock's options should probably confirm it.